macro/src/expand.rs

```diff
@@ -647,7 +650,7 @@ fn expand_cxx_function_shim(efn: &ExternFn, types: &Types) -> TokenStream {
     let call = if indirect_return {
         let ret = expand_extern_type(efn.ret.as_ref().unwrap(), types, true);
         setup.extend(quote_spanned! {span=>
-            let mut __return = ::cxx::core::mem::MaybeUninit::<#ret>::uninit();
+            let mut __return = ::cxx::core::mem::MaybeUninit::<#ret>::zeroed();
```

Comment:
This was related to the non-deterministic test failure you mentioned, right?

I think it'd be really good to understand this better, since it looks like this affects much more than just the `kj::Maybe` integration.